addons/website/static/src/active_elements/anchor_slide.js

```diff
@@ -0,0 +1,102 @@
+import { Component, useAttachedEl } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { scrollTo } from "@web_editor/js/common/scrolling";
+
+export class AnchorSlide extends Component {
+    static selector = "a[href^='/'][href*='#'], a[href^='#']";
+    static dynamicContent = {
+        "root": {
+            "t-on-click": "animateClick",
+        },
+    };
+    
+    setup() {
+        this.el = useAttachedEl();
+    }
+
+    /**
+     * @param {DOMElement} el the element to scroll to.
+     * @param {string} [scrollValue='true'] scroll value
+     * @returns {Promise}
+     */
+    scrollTo(el, scrollValue="true") {
+        scrollTo(el, {
+            duration: scrollValue === "true" ? 500 : 0,
+            extraOffset: this.computeExtraOffset(),
+        });
+    }
+    /**
+     * To be overridden.
+     */
+    computeExtraOffset() {
+        return 0;
+    }
+    /**
+     */
+    animateClick(ev) {
+        const ensureSlash = path => path.endsWith("/") ? path : path + "/";
+        if (ensureSlash(this.el.pathname) !== ensureSlash(window.location.pathname)) {
+            return;
+        }
+        // Avoid flicker at destination in case of ending "/" difference.
+        if (this.el.pathname !== window.location.pathname) {
+            this.el.pathname = window.location.pathname;
+        }
+        let hash = this.el.hash;
+        if (!hash.length) {
+            return;
+        }
+        // Escape special characters to make the jQuery selector to work.
+        hash = '#' + $.escapeSelector(hash.substring(1));
```

Comment:
jquery

Reply:
Added TODO comment about getting rid of JQ there: that escaping is non-trivial, and it might have to be added in a place like `dom.js` - but it is surprisingly only used in that one location.